learn/tanssi/external-security-providers/symbiotic.md

```diff
@@ -110,20 +112,133 @@ flowchart LR
+2. **Cross-Chain Messaging** - reward allocation data is sent using [XCM](https://docs.moonbeam.network/builders/interoperability/xcm/overview/) (Cross-Consensus Messaging), a standardized protocol for blockchain communication. [Snowbridge](https://wiki.polkadot.network/docs/learn-snowbridge) acts as a trustless bridge between Tanssi and Ethereum
+3. **`Gateway`** - once the message is relayed to the `Gateway` contract, this contract serves as Tanssi's authorized entry point on Ethereum for the Snowbridge bridge
+4. **`Middleware`** - the `Gateway` forwards the data to the [`Middleware`](https://github.com/moondance-labs/tanssi-symbiotic/blob/main/src/contracts/middleware/Middleware.sol), which is responsible for various tasks, including passing the information to the `OperatorReward` contract
+5. **`OperatorRewards`** - this is the final destination for the data. The [`OperatorRewards`](https://github.com/moondance-labs/tanssi-symbiotic/blob/main/src/contracts/rewarder/ODefaultOperatorRewards.sol) contract stores the Merkle tree of the reward allocations and handles the transfer of reward tokens when a claim is made
 
-!!! note
-    Slashing events can only be triggered by operators' misbehavings in the Tanssi Network itself. Tanssi networks, even if faulty or malicious, run in a sandboxed environment and can not cause slashing.
\ No newline at end of file
+```mermaid
+sequenceDiagram
+    participant Tanssi Network
+    participant Snowbridge (XCM)
+    participant Gateway
+    participant Middleware
+    participant OperatorRewards
+
+    Tanssi Network->>Tanssi Network: 1. Calculate rewards and generate Merkle root
+    Tanssi Network->>Snowbridge (XCM): 2. Send XCM message (Merkle root + data)
+    Snowbridge (XCM)->>Gateway: 3. Relay message
+    Gateway ->>Middleware: 4. Propagate rewards data
+    Middleware->>OperatorRewards: 5. distributeRewards()
+```
+
+#### Reward Claiming Phase {: #reward-claiming-phase }
+
+In the reward-claiming phase, operators and stakers can claim rewards based on their participation in the network. Tanssi determines the share distribution for operators and stakers, currently setting it at 20% for operators and 80% for stakers.
+
+1. **Operator Claims** - operators can claim their share by calling the `OperatorRewards` contract by using a cryptographic receipt that verifies their entitlement
+2. **Token Release** - the operator call triggers the token release, and the `OperatorRewards` sends the established amount to the operator
+3. **Token Distribution Stakers** - the remaining rewards are forwarded to the `StakerRewards` contract for further claiming of the staker
+4. **Staker Allocation** - the remaining 80% of the rewards are automatically directed to the [`StakerRewards`](https://github.com/moondance-labs/tanssi-symbiotic/blob/main/src/contracts/rewarder/ODefaultStakerRewards.sol) contract, where stakers can claim rewards proportional to their stake in the vaults
+
+```mermaid
```

Comment:
Black lines in this diagram too that should be updated

Reply:
Also same comment about it being unclear why the top row and bottom row are the exact same